src/main/java/com/github/princesslana/eriscasper/data/resource/AuditLogChange.java

```diff
@@ -0,0 +1,163 @@
+package com.github.princesslana.eriscasper.data.resource;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.github.princesslana.eriscasper.data.Data;
+import com.github.princesslana.eriscasper.data.DataException;
+import com.github.princesslana.eriscasper.data.Snowflake;
+import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+
+@SuppressWarnings("unused")
+public class AuditLogChange<T> {
+
+  private Class<T> valueClass;
+  private T newValue;
+  private T oldValue;
+  private String key;
```

Comment:
Shouldn't it be possible to make this class immutable and all these fields final?

Reply:
Yeah, I think we can do that, and T is useful just for checking the instance of it being an array, but I could maybe change it Immutable, I'll try and see what turns up.